markdown.go

```diff
@@ -241,6 +241,16 @@ func (p *parser) getRef(refid string) (ref *reference, found bool) {
 	return ref, found
 }
 
+func (p *parser) isFootnote(ref *reference) bool {
+	for _, v := range p.notes {
```

Comment:
I guess in order to do that we need to change `p.notes` from a slice to a map in order to avoid `n^2` complexity.